sqlmesh/core/schema_diff.py

```diff
@@ -655,7 +658,10 @@ def compare_structs(
         """
         return [
             op.expression(table_name, self.array_element_selector)
-            for op in self._from_structs(current, new)
+            for op in self._from_structs(
+                normalize_identifiers(current, dialect=self.dialect),
+                normalize_identifiers(new, dialect=self.dialect),
```

Comment:
I'm skeptical about this section, I think we can end up normalizing identifiers twice: `current` and `new` are built off of columns-to-types mappings [here](https://github.com/TobikoData/sqlmesh/blob/63e2dbd54430ca16682749ca1c2e5c5f71312303/sqlmesh/core/schema_diff.py#L677-L679). For reference, [here](https://github.com/TobikoData/sqlmesh/blob/63e2dbd54430ca16682749ca1c2e5c5f71312303/sqlmesh/core/plan/builder.py#L454-L464) is one place where `compare_columns` is called. The problem I'm seeing is that the `columns_to_types` property in models doesn't preserve quotes, it simply stores their normalized names.

For example, if I change the example project's seed model to:

```sql
MODEL (
    name sqlmesh_example.seed_model,
    kind SEED (
      path '../seeds/seed_data.csv'
    ),
    columns (
      id INTEGER,
      "ItemId" INTEGER,
      event_date DATE
    ),
    grain (id, event_date)
  );
```

Then observe how `columns_to_types` looks:

```python
>>> ctx.models['"db"."sqlmesh_example"."seed_model"'].columns_to_types
{'id': DataType(this=Type.INT, nested=False), 'ItemId': DataType(this=Type.INT, nested=False), 'event_date': DataType(this=Type.DATE, nested=False)}
```

Notice how there are no quotes around `ItemId` in the mapping. If I'm reading the control flow correctly, you'll force lowercase/uppercase names that are already normalized here, even when you shouldn't, which can cause its own problems.

Let me know if I'm missing something, but I'd double check this before merging, and make sure to test this scenario.

Reply:
Yes you're right I tested with various engined and the columns_to_types are already normalised based on case sensitivity. So using `normalize_identifiers` is incorrectly forcing normalization when it shouldn't here. Also, in the other section that `compare_columns` is used: 

https://github.com/TobikoData/sqlmesh/blob/63e2dbd54430ca16682749ca1c2e5c5f71312303/sqlmesh/core/engine_adapter/base.py#L904-L907

the columns are fetched from the database, which again include the correct case sensitivity (so if a column for example "ItemId" is quoted, it will be retrieved as ItemId). So the SchemaDiffer does account for case sensitivity, operating under the assumption that identifiers are already correctly normalized when the columns are passed to `compare_columns`